packages/eas-cli/src/commands/fingerprint/compare.ts

```diff
@@ -0,0 +1,378 @@
+      } else if (diff.op === 'changed') {
+        sourceType = diff.beforeSource.type;
+      }
+      return sourceType === 'contents';
+    });
+
+    for (const diff of contentDiffs) {
+      printContentDiff(diff);
+    }
+  }
+}
+
+function printContentDiff(diff: FingerprintDiffItem): void {
+  if (diff.op === 'added') {
+    const sourceType = diff.addedSource.type;
+    if (sourceType === 'contents') {
+      printContentSource({
+        op: diff.op,
+        sourceType,
+        contentsId: diff.addedSource.id,
+        contentsAfter: diff.addedSource.contents,
+      });
+    }
+  } else if (diff.op === 'removed') {
+    const sourceType = diff.removedSource.type;
+    if (sourceType === 'contents') {
+      printContentSource({
+        op: diff.op,
+        sourceType,
+        contentsId: diff.removedSource.id,
+        contentsBefore: diff.removedSource.contents,
+      });
+    }
+  } else if (diff.op === 'changed') {
+    const sourceType = diff.beforeSource.type;
+    if (sourceType === 'contents') {
+      if (diff.afterSource.type !== 'contents') {
+        throw new Error(
+          `Changed fingerprint source types must be the same, received ${diff.beforeSource.type}, ${diff.afterSource.type}`
+        );
+      }
+      printContentSource({
+        op: diff.op,
+        sourceType: diff.beforeSource.type, // before and after source types should be the same
+        contentsId: diff.beforeSource.id, // before and after content ids should be the same
+        contentsBefore: diff.beforeSource.contents,
+        contentsAfter: diff.afterSource.contents,
+      });
+    }
+  }
+}
+
+function getDiffFilePathFields(diff: FingerprintDiffItem): FormatFieldsItem | null {
+  if (diff.op === 'added') {
+    const sourceType = diff.addedSource.type;
+    if (sourceType !== 'contents') {
+      return getFilePathSourceFields({
+        op: diff.op,
+        sourceType,
+        filePath: diff.addedSource.filePath,
+      });
+    }
+  } else if (diff.op === 'removed') {
+    const sourceType = diff.removedSource.type;
+    if (sourceType !== 'contents') {
+      return getFilePathSourceFields({
+        op: diff.op,
+        sourceType,
+        filePath: diff.removedSource.filePath,
+      });
+    }
+  } else if (diff.op === 'changed') {
+    const sourceType = diff.beforeSource.type;
+    if (sourceType !== 'contents') {
+      return getFilePathSourceFields({
+        op: diff.op,
+        sourceType: diff.beforeSource.type, // before and after source types should be the same
+        filePath: diff.beforeSource.filePath, // before and after filePaths should be the same
+      });
+    }
+  }
+  return null;
+}
+
+function getFilePathSourceFields({
+  op,
+  sourceType,
+  filePath,
+}: {
+  op: 'added' | 'removed' | 'changed';
+  sourceType: 'dir' | 'file';
+  filePath: string;
+}): FormatFieldsItem {
+  if (sourceType === 'dir') {
+    if (op === 'added') {
+      return { label: 'new directory:', value: filePath };
```

Comment:
in the example output from the op it looks like there were two `:` in the output - "new directory::` - it'd be good to double check that so we only have one.